### PR TITLE
Markdown lint support

### PR DESCRIPTION
**Pull request by superm1**
So I think it's going to be a lot of busy work fixing up the rest, but this is probably the way to go to keep codacity analysis happy.

Type of pull request:
- [ ] New plugin (Please include [new plugin checklist](https://github.com/fwupd/fwupd/wiki/New-plugin-checklist))
- [ ] Code fix
- [ ] Feature
- [x] Documentation

### CODE REVIEW

**Comment by superm1** [comment]
Pre-commit CI goes from 20-30s up to 1m due to setup.  I'm not sure if maybe https://github.com/DavidAnson/markdownlint-cli2 is better, might be faster.

**Comment by superm1** [comment]
Also, can we maybe turn off codacity looking at markdown?  I'm not sure we want to be so pedantic to their style.  I'd rather we control it to our own that we can have a conf file for (like the pre-commit I did, but maybe we still need more tweaking)

**Comment by hughsie** [comment]
> Pre-commit CI goes from 20-30s up to 1m due to setup

I think that's fine.

> Also, can we maybe turn off codacity looking at markdown?

I think if it's being checked in precommit I think that's fine. I think we should turn off `MD013/line-length Line length` too.

**Comment by FFY00** [comment]
> Pre-commit CI goes from 20-30s up to 1m due to setup.

But this is just in CI, which should be fine. Locally, pre-commit will only check modified files if you have the hook.

**Comment by hughsie** [comment]
@superm1 you happy with the changes? I hope it's okay I fixed up the remaining issues.

File: .markdownlint.yaml [hunk 1]
```diff
@@ -0,0 +1,2 @@
+MD013:
+  line_length: 1000
```
Comment (superm1) [inline .markdownlint.yaml:2]: I don't think we need both yaml and json, it should be one or the other.

File: .pre-commit-config.yaml [hunk 2]
```diff
@@ -48,3 +48,8 @@ repos:
       language: system
       entry: shellcheck --severity=error -e SC2068
       types: [shell]
+-   repo: https://github.com/igorshubovych/markdownlint-cli
+    rev: v0.27.1
+    hooks:
+    - id: markdownlint
+      args: ['--fix', '--config', '.markdownlint.yaml', '--ignore', '.github']
```
Comment (superm1) [inline .pre-commit-config.yaml:55]: I don't think you need --config if you use json.
Also --ignore on .github is that right?

**Review by superm1 — COMMENTED** [review]
Just a nit on the configuration so we don't end up with two copies in different formats

File: .markdownlint.yaml (hunk 1 above)
Comment (hughsie) [reply .markdownlint.yaml]: TIL about the existing json file. I've removed the YAML one.

File: .pre-commit-config.yaml (hunk 2 above)
Comment (hughsie) [reply .pre-commit-config.yaml]: The .github one is required as the pull_request_template.md isn't really a proper document; it has no title and some of the rules don't apply. If you _want_ to make it a proper document that's fine too.

Comment (FFY00) [reply .pre-commit-config.yaml]: Btw, you can exclude directly in pre-commit, which will prevent pre-commit from trying to run the hook on those files.

https://github.com/pypa/build/blob/5a3c5243a2135236f207da3a4b6401e1561f952d/.pre-commit-config.yaml#L11

Comment (hughsie) [reply .pre-commit-config.yaml]: oh, specific to markdownlint-cli? Got an example?

Comment (superm1) [reply .pre-commit-config.yaml]: We run on --all-files for CI. So I think the exclusion you're doing is fine

**Pull request merged**